### PR TITLE
Sdc gallery app, boolean choice workflow.

### LINKED ISSUE #992: Sdc gallery app boolean choice component workflow
**Is your feature request related to a problem? Please describe.**
1. Design boolean choice questionnaire json  without required fields. (error message will not be shown as field is not mandatory)
2. Api to retrieve single choice questionnaire json from model view.
3. Handle single choice item click
4. Implement questionnaire fragment and model view to show questionnaire json
5. Show action bar, update title as per screen, align title text.
6. Update color palette hex codes.

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Would you like to work on the issue?**
Please state if this issue should be assigned to you or who you think could help to solve this issue.

### PR DESCRIPTION
**Pull request by santosh-pingle**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #992 

**Description**
Design boolean choice questionnaire json with without required fields. (error message will not be shown as field is not mandatory)
Api to retrieve single choice questionnaire json from model view.
Handle single choice item click
Implement questionnaire fragment and model view to show questionnaire json
Show action bar, update title as per screen, align title text.
Update the color palettes.

**Alternative(s) considered**
Have you considered any alternatives? And if so, why have you chosen the approach in this PR?

**Type**
Choose one: Feature

**Screenshots (if applicable)**

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md)
- [x] I have read [How to Contribute](https://github.com/google/android-fhir/blob/master/docs/contributing.md)
- [x] I have read the [Developer's guide](https://github.com/google/android-fhir/wiki/Developer's-Guide)
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate )
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally
- [x] I have built and run the reference app(s) to verify my change fixes the issue and/or does not break the reference app(s)

### CODE REVIEW

**Comment by santosh-pingle** [comment]
Boolean choice componnet.
Screenshot_20220105-124404 (1)

Boolean choice questionnaire
Screenshot_20220105-124535 (1)

File: catalog/src/main/java/com/google/android/fhir/catalog/ComponentListFragment.kt [hunk 1]
```diff
@@ -32,10 +35,49 @@ class ComponentListFragment : Fragment(R.layout.fragment_components) {
...
+    launchQuestionnaireFragment(component)
+  }
+
+  private fun launchQuestionnaireFragment(component: ComponentListViewModel.Component) {
+    findNavController()
+      .navigate(
+        ComponentListFragmentDirections.actionComponentsFragmentToGalleryQuestionnaireFragment(
+          context?.getString(component.textId) ?: "",
+          viewModel.getQuestionnaire(component),
+          null
```
Comment (davegwatson) [inline catalog/src/main/java/com/google/android/fhir/catalog/ComponentListFragment.kt:79]: can we document what this null parameter is (/* paramName=*/ null) or update the method in question to have null as the default for this parameter?

File: catalog/src/main/java/com/google/android/fhir/catalog/ComponentListViewModel.kt [hunk 2]
```diff
@@ -29,8 +29,16 @@ class ComponentListViewModel(application: Application, private val state: SavedS
     return Component.values().toList()
   }
 
+  fun getQuestionnaire(component: Component): String {
+    return when (component) {
+      Component.BOOLEAN_CHOICE -> "single_choice_questionnaire_boolean.json"
+      else -> "" // TODO remove else when all components cases are added to the when.
```
Comment (davegwatson) [inline catalog/src/main/java/com/google/android/fhir/catalog/ComponentListViewModel.kt:35]: Issue# for this TODO as well?

**Review by davegwatson — APPROVED** [review]
This is my first review - I don't really know what i'm doing yet on this project (or with Kotlin), but everything generally LGTM.

File: catalog/src/main/java/com/google/android/fhir/catalog/ComponentListViewModel.kt (hunk 2 above)
Comment (santosh-pingle) [reply catalog/src/main/java/com/google/android/fhir/catalog/ComponentListViewModel.kt:35]: #1076

File: catalog/src/main/java/com/google/android/fhir/catalog/ComponentListFragment.kt [hunk 3]
```diff
@@ -32,10 +35,49 @@ class ComponentListFragment : Fragment(R.layout.fragment_components) {
...
+      ComponentsRecyclerViewAdapter(::onItemClick).apply {
+        submitList(viewModel.getComponentList())
+      }
     val recyclerView = requireView().findViewById<RecyclerView>(R.id.componentsRecyclerView)
     recyclerView.adapter = adapter
     recyclerView.layoutManager = GridLayoutManager(requireContext(), 2)
   }
+
+  private fun onItemClick(component: ComponentListViewModel.Component) {
+    // TODO Remove check when all components questionnaire json are updated.
```
[the remark this answers is not in the document]
Comment (santosh-pingle) [reply catalog/src/main/java/com/google/android/fhir/catalog/ComponentListFragment.kt:66]: #1076

File: catalog/src/main/java/com/google/android/fhir/catalog/ComponentListViewModel.kt [hunk 4]
```diff
@@ -29,8 +29,17 @@ class ComponentListViewModel(application: Application, private val state: SavedS
     return Component.values().toList()
   }
 
+  fun getQuestionnaire(component: Component): String {
+    return when (component) {
+      Component.BOOLEAN_CHOICE -> "single_choice_questionnaire_boolean.json"
```
Comment (jingtang10) [inline catalog/src/main/java/com/google/android/fhir/catalog/ComponentListViewModel.kt:34]: i think a follow up PR should make the file name a field in the `enum class Component`

File: catalog/src/main/java/com/google/android/fhir/catalog/GalleryQuestionnaireViewModel.kt [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+package com.google.android.fhir.catalog
+
+import android.app.Application
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.withContext
+
+class GalleryQuestionnaireViewModel(application: Application, private val state: SavedStateHandle) :
```
Comment (jingtang10) [inline catalog/src/main/java/com/google/android/fhir/catalog/GalleryQuestionnaireViewModel.kt:25]:
```suggestion
class DemoQuestionnaireViewModel(application: Application, private val state: SavedStateHandle) :
```

File: catalog/src/main/res/layout/fragment_gallery_questionnaire.xml [hunk 6]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="utf-8" ?>
```
Comment (jingtang10) [inline catalog/src/main/res/layout/fragment_gallery_questionnaire.xml:1]: change the name from gallery to demo.

**Review by jingtang10 — APPROVED** [review]

File: catalog/src/main/java/com/google/android/fhir/catalog/ComponentListViewModel.kt (hunk 4 above)
Comment (santosh-pingle) [reply catalog/src/main/java/com/google/android/fhir/catalog/ComponentListViewModel.kt]: #1079

**Pull request merged**